### ISSUE TITLE
Notifying deleted objects

### ISSUE THREAD

**Issue by Mateasek** [cherab/core#215]
https://github.com/cherab/core/blob/d58c8c26470209be9643a8c23b9dc9385bbe30e9/cherab/core/utility/notify.py#L95-L102

Recently I ran into a problem, which I traced to the notifier object. It should detect references to deleted objects and remove them, but I still got `AttributeError: 'NoneType' object has no attribute ....`.

Shouldn't the referenced piece of code test the existence before getting the method? 
``` python
             # bound method
                instance = reference[0]()

                # does the object still exist
                if instance is None:
                    dead_callbacks.append(reference)
                    continue

                method = instance.__getattribute__(reference[1])
```

**Comment by Mateasek** [comment]
If this is a bug, I've created PR #217 with the fix.

**Comment by CnlPepper** [comment]
This is unlikely to be due to the notifier and more likely to be related to a malformed object, possibly one where a method is being dynamically deleted or accidentally set to None. I can't debug this without the calling code.

**Comment by Mateasek** [comment]
I got it when working on segmenting laser for Thomon scattering model. I will try to put together the setup to let you test it. Still, I would like to ask, Is there a point of getting the method if the instance is None?

**Comment by CnlPepper** [comment]
It should never get to that point, implying something else has gone wrong before that stage.

**Comment by CnlPepper** [comment]
We could change the order to perform the instance check first, which would be ok, but it's a sensitive part of cherab so I'd want to be sure we need to make the change.

**Comment by Mateasek** [comment]
I understand. I will try to see if this could have been cause by me mistreating something in the TS model. Also I will try to replicate this issue with some other parts than TS model,  if I can't I will try to get to you the version of the model causing the error and script I was running. Thanks @CnlPepper

**Comment by CnlPepper** [comment]
Sorry, ignore my previous rambling. I've had a closer look at this, you were right this was a mistake.  I've fixed it in 686e45d.
